api/api.go

```diff
@@ -0,0 +1,190 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/contiv/errored"
+	"github.com/contiv/volplugin/config"
+	"github.com/contiv/volplugin/errors"
+	"github.com/contiv/volplugin/lock"
+	"github.com/contiv/volplugin/storage/control"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// VolumeRequest is taken from
+// https://github.com/calavera/docker-volume-api/blob/master/api.go#L23
+type VolumeRequest struct {
```

Comment:
Ah ok; I will retool to use those then.

On 25 Jun 2016, at 8:07, Madhav Puri wrote:

>> +	"net/http"
>> +	"os"
>> +	"strings"
>> +
>> +	"github.com/contiv/errored"
>> +	"github.com/contiv/volplugin/config"
>> +	"github.com/contiv/volplugin/errors"
>> +	"github.com/contiv/volplugin/lock"
>> +	"github.com/contiv/volplugin/storage/control"
>> +
>> +	log "github.com/Sirupsen/logrus"
>> +)
>> +
>> +// VolumeRequest is taken from
>> +// 
>> https://github.com/calavera/docker-volume-api/blob/master/api.go#L23
>> +type VolumeRequest struct {
>
> I think docker has these structures defined as well 
> (https://github.com/docker/docker/blob/master/vendor/src/github.com/docker/engine-api/types/types.go#L437), 
> may be we should use those or vendor them in to avoid any 
> incompatibilities between structs.
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/313/files/dd6c3cf3fdc8e0f0c1b54f2f638b5f131ae0c657#r68489642